lib/shopify_app/controller_concerns/with_shopify_id_token.rb

```diff
@@ -5,31 +5,30 @@ module WithShopifyIdToken
     extend ActiveSupport::Concern
 
     def shopify_id_token
-      @shopify_id_token ||= id_token_from_request_env || id_token_from_authorization_header || id_token_from_url_param
+      @shopify_id_token ||= id_token_from_authorization_header || id_token_from_url_param
+    end
+
+    def jwt_payload
+      @jwt_payload ||= shopify_id_token.present? ? ShopifyAPI::Auth::JwtPayload.new(shopify_id_token) : nil
     end
 
     def jwt_shopify_domain
-      request.env["jwt.shopify_domain"]
+      @jwt_shopify_domain ||= jwt_payload.present? ? ShopifyApp::Utils.sanitize_shop_domain(jwt_payload.dest) : nil
     end
 
     def jwt_shopify_user_id
-      request.env["jwt.shopify_user_id"]
+      jwt_payload&.sub&.to_i
```

Comment:
I think we can just use `jwt_payload&.shopify_user_id` here (and `jwt_payload&.shopify_domain`, `jwt_payload&.expire_at` ). Similar to how the [middleware used to behave](https://github.com/Shopify/shopify_app/pull/1861/files#diff-f5062bb2e1d5a0819d14760dd0ce26827465e4811baa9ce69736b22c6f0b31fdL41). These helpers had been added to [ShopifyAPI::Auth::JwtPayload](https://github.com/Shopify/shopify-api-ruby/commit/6c9c2f80628dcea0e3ff33a42b8c61023747981b) to avoid having to parse these everytime.

Reply:
Nice, I didn't realize these were available. Updated.